client/src/main/java/org/apache/rocketmq/client/impl/consumer/DefaultLitePullConsumerImpl.java

```diff
@@ -860,9 +861,16 @@ public class PullTaskImpl implements Runnable {
         private final MessageQueue messageQueue;
         private volatile boolean cancelled = false;
         private Thread currentThread;
+        private final CommunicationMode communicationMode;
 
         public PullTaskImpl(final MessageQueue messageQueue) {
             this.messageQueue = messageQueue;
+            communicationMode = CommunicationMode.SYNC;
+        }
+
+        public PullTaskImpl(final MessageQueue messageQueue, CommunicationMode communicationMode) {
+            this.messageQueue = messageQueue;
+            this.communicationMode = communicationMode;
```

Comment:
Good job! How do users set whether the pullTask is asynchronous or synchronous?

Reply:
Thanks for your comment! I have the same confusion about the entrance of asynchronous pullTask.

As you see, I create another constructor here. When user creates the PullTaskImpl with "Async", the pullTask will run in the asynchronous way. However, I find that the PullTaskImpl only be created in other methods (e.g. seek(), startPullTask()) in this class. 
If we let user decide which way to pull the tasks, we need to add a new parameter "communicationMode" for all upstream functions **or** in the class DefaultLitePullConsumerImpl (maybe better? just a private parameter with a set() function). I think either of them will make the code more complex.

This is my thought on the entrance design, but as I'm a freshman in RocketMQ, there may be some areas where I have not considered thoroughly. I would appreciate your advice and guidance on this.

Reply:
You are right. It is not appropriate for users to decide whether to synchronize or asynchronously. IMO, we should replace the current synchronous pull implementation after the asynchronous pull is stable.

Reply:
Thank you for your approval! Based on your suggestion, I think we can keep this constructor to facilitate our testing of asynchronous pull. Besides, what else can I do for this PR?

Reply:
> Thank you for your approval! Based on your suggestion, I think we can keep this constructor to facilitate our testing of asynchronous pull. Besides, what else can I do for this PR?

Go ahead, you can fix the problem of the comments, then we should do some tests. :-)

Reply:
> 

I have fixed the problem mentioned in comments. Thanks for your effort! Please let me know if there is anything I can do for it.